docs/docs/api/websocket.md

```diff
@@ -20,93 +20,71 @@ To execute the handshake with `/ws.communication` you must set an
 `Authorization` header where the value is the authorization token obtained [from
 the API](/api/introduction#authentication).
 
-## Outbound Queues
+## Event Types
 
-Outbound queues follow the pattern `/queue/:event_type[/:action}]` and deliver
-JSON encoded payloads.
+All event updates are sent to the `/events` queue as JSON encoded payloads. The `type`
+field informs the client of the kind of update that is encoded in the payload.
 
 ### Message
 
-`/queue/message`
-
-Incoming payloads notify connected clients that a message was created or
-updated.
-
-**Sample payload**
-
 ```json5
 {
-  "conversation_id": "{UUID}",
-  "channel_id": "{UUID}",
-  "message": {
-    "id": "{UUID}",
-    "content": '{"text":"Hello World"}',
-    // source message payload
-    "delivery_state": "{String}",
-    // delivery state of message, one of PENDING, FAILED, DELIVERED
-    "sender_type": "{string/enum}",
-    // See glossary
-    "sent_at": "{string}",
-    //'yyyy-MM-dd'T'HH:mm:ss.SSSZ' date in UTC form, to be localized by clients
-    "source": "{String}"
-    // one of the possible sources
+  "type": "message",
+
+  "payload": {
+    "conversation_id": "{UUID}",
+    "channel_id": "{UUID}",
+    "message": {
+      "id": "{UUID}",
+      "content": '{"text":"Hello World"}',
+      // source message payload
+      "delivery_state": "pending|failed|delivered",
+      // delivery state of message, one of pending, failed, delivered
+      "sender_type": "{string/enum}",
+      // See glossary
+      "sent_at": "{string}",
+      //'yyyy-MM-dd'T'HH:mm:ss.SSSZ' date in UTC form, to be localized by clients
+      "source": "{String}"
+      // one of the possible sources
+    }
   }
 }
 ```
 
-### Unread count
-
-`/queue/unread-count`
-
-Incoming payloads notify connected clients of the unread message count for a
-specific conversation at the time of delivery. Clients should keep track of the
-latest time the unread count for a specific conversation was updated and update
-the value only for a more recent count.
+### Metadata
 
 **Sample payload**
 
 ```json5
 {
-  conversation_id: "{UUID}",
-  //unique conversation id
-  unread_message_count: 42,
-  //the number of unreaded messages in this conversation
-  timestamp: "{string}"
-  //'yyyy-MM-dd'T'HH:mm:ss.SSSZ' date in UTC form, to be localized by clients
+  "type": "metadata",
```

Comment:
@lucapette and @paulodiniz So there are two ways of doing this:

- Send every single metadata record that we get immediately using a consumer group (advantage: very fast)

- Aggregate metadata to a map and emit complete metadata objects on every websocket event (advantage: clients always get the latest metadata document state, disadvantage: slower, can't use a consumer group)

What are your thoughts? Locally I have the former version, but I feel like the latter will need to a nicer API for clients.

Reply:
Resolved offline: The latter approach is better for the reasons listed above